### DISCUSSION TITLE
Speeding up MIS simulations

### DISCUSSION THREAD

**Discussion by lhenriet** [pasqal-io/Pulser#260] [Ideas]
Hello everyone. MIS simulations are slow in pulser. However, there might be a way to increase the simulation speed by using a truncated Hilbert space. Ideas about how/when we could add this feature into pulser ? @LucasGitQ @cdalyac @sebgrijalva

**Comment by HGSilveri** [comment]
Would this be a feature used only on MIS? From my limited understanding, the THS needs some particular constraints to be fulfilled in order to give a good approximation to the regular simulation results. If we make it available for every Sequence and Register, we'll need to verify that all the requirements are met beforehand. Is this feasible?

If, on the other hand, we are just interested in using this on the MIS (or other specific applications), I think it would go well together with the idea of having an higher level class that would take in the graph on which to find the MIS and spit out the best Register configuration and Sequence. In this case, we would control things from the start and would be able to tell whether the THS can be used or not.

**Reply by lhenriet** [reply]
In principle, THS could always be used on the ground rydberg transition. But because of the hurdles you mention, I would be in favor of using it specifically for MIS (and implementing pre-processing routines there, as you suggest).

**Reply by cdalyac** [reply]
I agree with Henrique, option 2 seems more natural. 

On our side, we should try to study more thoroughly the compared dynamics between THS and Pulser in order to come up with necessary / sufficient conditions. If we can come up with a complete list of requirements it could then be implemented on any g-r transition.
